projects/core/src/cart/connectors/delivery/cart-delivery.adapter.ts

```diff
@@ -0,0 +1,62 @@
+import { Observable } from 'rxjs';
+import { Address, DeliveryMode } from '../../../occ/occ-models/occ.models';
+
+export abstract class CartDeliveryAdapter {
+  /**
+   * Abstract method used to create address in cart
+   *
+   * @param userId
+   * @param cartId
+   * @param address
+   */
+  abstract createAddress(
+    userId: string,
+    cartId: string,
+    address: Address
+  ): Observable<Address>;
+
+  /**
+   * Abstract method used to set adress for delivery
+   *
+   * @param userId
+   * @param cartId
+   * @param addressId
+   */
+  abstract setAddress(
+    userId: string,
+    cartId: string,
+    addressId: string
+  ): Observable<any>;
+
+  /**
+   * Abstract method used to set delivery mode on cart
+   *
+   * @param userId
+   * @param cartId
+   * @param deliveryModeId
+   */
+  abstract setMode(
+    userId: string,
+    cartId: string,
+    deliveryModeId: string
```

Comment:
Can we make the deliveryMode tye safe? (optionally also allow for string)?

Reply:
Hmm... not sure. We get delivery modes from backend, so it looks that they are backend driven and code could be any string.